src/duration/mod.rs

```diff
@@ -120,8 +123,13 @@ impl FromStr for Duration {
             // in 1.3.0 but u128 was only added in 1.26.0. See
             // - https://users.rust-lang.org/t/why-duration-as-from-millis-uses-different-primitives/89302
             // - https://github.com/rust-lang/rust/issues/58580
-            duration +=
-                std::time::Duration::from_millis((value * unit.millis()).try_into().unwrap());
+            match duration.checked_add(std::time::Duration::from_millis(
+                (value * unit.millis()).try_into().unwrap(),
```

Comment:
Imagine the input `5000y`. Then `value * unit.millis()` would also fail, is that correct?
In that case we would need a `checked_mul` here as well. I'm happy to re-use the error variant and just say `The duration "<original-string, e.g. 5000y3d12h>" is too long`

Reply:
Checked in 25e0671